base/applications/screensavers/logon/logon.c

```diff
@@ -18,159 +18,109 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include <stdarg.h>
 #include <windef.h>
 #include <winbase.h>
 #include <wingdi.h>
 #include <winuser.h>
 #include <scrnsave.h>
 #include <stdlib.h>
-#include <tchar.h>
 
 #include "resource.h"
 
-#define RANDOM( min, max ) ((rand() % (int)(((max)+1) - (min))) + (min))
+#define RANDOM(min, max) ((rand() % (int)(((max)+1) - (min))) + (min))
 
-#define APPNAME               _T("Logon")
 #define APP_TIMER             1
-#define APP_TIMER_INTERVAL    2000
+#define APP_TIMER_INTERVAL    10000
 
-HBITMAP
-GetScreenSaverBitmap(VOID)
+HBITMAP GetScreenSaverBitmap(VOID)
 {
-    OSVERSIONINFOEX osvi;
-
-    ZeroMemory(&osvi, sizeof(OSVERSIONINFOEX));
-    osvi.dwOSVersionInfoSize = sizeof(OSVERSIONINFOEX);
-    GetVersionEx ((OSVERSIONINFO *) &osvi);
-
-    switch(osvi.wProductType)
-    {
-        case VER_NT_WORKSTATION:
-            return LoadImage(GetModuleHandle(NULL), MAKEINTRESOURCE(IDB_WORKSTATION), IMAGE_BITMAP, 0, 0, LR_CREATEDIBSECTION);
-            break;
-        default:
-            return LoadImage(GetModuleHandle(NULL), MAKEINTRESOURCE(IDB_SERVER), IMAGE_BITMAP, 0, 0, LR_CREATEDIBSECTION);
-            break;
-    }
+    return LoadImage(GetModuleHandle(NULL), MAKEINTRESOURCE(IDB_LOGO), IMAGE_BITMAP, 0, 0, LR_CREATEDIBSECTION);
 }
 
-LRESULT
-CALLBACK
-ScreenSaverProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
+LRESULT CALLBACK ScreenSaverProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
     static RECT rect;
     static HBITMAP bitmap;
 
-    switch (message)
+    switch (uMsg)
     {
         case WM_CREATE:
         {
-            bitmap = GetScreenSaverBitmap ();
+            bitmap = GetScreenSaverBitmap();
             if (bitmap == NULL)
             {
+                /* Extremely unlikely, message not localized. */
                 MessageBox(hWnd,
-                           _T("Fatal Error: Could not load bitmap"),
-                           _T("Error"),
+                           L"Fatal Error: Could not load bitmap",
```

Comment:
If you use explicit unicode strings there, please change all unicode functions and types to use the explicit W name (e.g. MessageBoxW, WCHAR, etc.)

Reply:
Done. Though I will leave this unresolved if you want to dispute `LoadImageW` or `GetObjectW`

Reply:
I don't think there has to be any dispute for those functions.